SWIGGYDEMO_REACT/swiggy-react-app/src/components/itemDetailsAndCart/components/listByCategory/ListbyCategory.js

```diff
@@ -0,0 +1,162 @@
+import './listByCategory.style.css';
+
+import React, { useEffect, useState } from "react";
+import Items from "./components/items";
+
+function ListByCategory ({data, category, cartItems, onClickADD, onClickPlus, onClickMinus, isVeg, searchInputText, onFilter}) {
+
+    const [ searchResultItems, setSearchResultItems ] = useState([]);
+
+    const [ categoryListForSearch, setCategoryListForSearch ] = useState([category]);
+
+    const [ categoryListForFilter, setCategoryListForFilter ] = useState([category]);
+
+    let arrayOfItemsOrderedByCategory = [];
+
+    category.forEach( (i, index) => {
+        let arrayOfItems = [];
+        arrayOfItems = data.filter( (d) => {
+            return d.category === i;
+        } )
+        arrayOfItemsOrderedByCategory[index] = arrayOfItems;
+    })
+
+    const [ itemsOrderedByCategory, setItemsOrderedByCategory ] = useState([...arrayOfItemsOrderedByCategory])
+
+    function updateItems () {
+        let arrayOfSearchedItems = [];
```

Comment:
try to refactor without let, using const

Reply:
Used const